code/modules/shuttle/shuttle.dm

```diff
@@ -471,9 +471,15 @@
 	var/list/turfs = ripple_area(S1)
 	for(var/t in turfs)
 		ripples += new /obj/effect/temp_visual/ripple(t, animate_time)
+		for(var/obj/machinery/door/airlock/A in range(1,src))
+			A.danger_lights = TRUE
+			A.update_icon()
```

Comment:
You're calling update_icon three times per airlock here. That's three times as much CPU time taken just to flash warning lights on airlocks every single time a shuttle arrives.

Reply:
The proper way to do this would be to check the shuttle for air locks next to non-shuttle turfs and look for any airlocks on the other side.

There already exist code that does this to some degree in shuttles. its used to close any docking doors.

Very likely what we should do is make a function on the stationary that takes a mobile and finds any connecting docking doors.

@ninjanomnom @coiax you guys know these systems best